WMR RD View Setup

Mrand03 opened the pull request:
Changes made in PR:

- Changed the housing index.jsx to go to the RD View only if you are faculty ( We can change this when we make a group for RD's). 
- Created an RDView folder within the housing/components folder. This new folder is where all of the RDView components will go as well as where the index.jsx for the RD View is.
- Created a card to edit documents in he RD View which has links to SharePoint and our room range program, as well as a dialog box that pops up when you click the edit button next to "RA/AC Task List". The code for the edit button is in a folder within RDView called components which will hold any components needed for the RD View.
- The code to edit tasks is just the dialog box that pops up, you cannot type anything yet.

dli505 commented on src/views/Housing/index.jsx:
```diff
@@ -1,9 +1,22 @@
-  </Grid>
-);
+const Housing = () => {
+  const isFaculty = useAuthGroups(AuthGroup.Faculty);
```
I have an `isFacStaff` function that we can import...
`import { Profile as profileType, isFacStaff as checkIsFacStaff } from 'services/user';`
Hopefully this works?

dli505 commented on src/views/Housing/index.jsx:
```diff
@@ -1,9 +1,22 @@
+const Housing = () => {
+  const isFaculty = useAuthGroups(AuthGroup.Faculty);
+
+  if (isFaculty) {
```
If it does work then this would be `isFacStaff` instead I think

Again not sure if it does work but worth a try? If not then we can use the auth groups

dli505 commented on src/views/Housing/components/RDView/updateTasks/index.jsx:
```diff
@@ -0,0 +1,27 @@
+  const [open, setOpen] = useState(false);
+
+  return (
+    <div>
```
I think in React instead of using divs we just use empty jsx fragments, like
`<></>`

dli505 reviewed (changes requested):
Yay Mya this looks really great! I added some quick comments, and if you could paste a screenshot of what the screen looks like that would look really awesome!

Mrand03 commented:
Hi Danya,

Thank you for your suggestions! I changed how we check if the user is faculty to what you suggested and it worked so we can use it. I also changed the div to <> </> and it caused no issues. Let me know if you have any other suggestions!

dli505 reviewed (approved):
Looks good to me!
